easybuild/tools/filetools.py

```diff
@@ -408,7 +416,18 @@ def extract_file(fn, dest, cmd=None, extra_options=None, overwrite=False, forced
 
     run.run_cmd(cmd, simple=True, force_in_dry_run=forced)
 
-    return find_base_dir()
+    # note: find_base_dir also changes into the base dir!
+    base_dir = find_base_dir()
+
+    # if changing into obtained directory is not desired,
+    # change back to where we came from (unless that was a non-existing directory)
+    if not change_into_dir:
+        if cwd is None:
+            raise EasyBuildError("Can't change back to non-existing directory after extracting %s in %s", fn, dest)
```

Comment:
We don't have to the path to the non-existing directory, since you can't determine it...

```
$ mkdir /tmp/thiswontbeherelong
$ cd /tmp/thiswontbeherelong
$ python -c 'import os; print(os.getcwd())'
/private/tmp/thiswontbeherelong

$ rm -rf $PWD
$ python -c 'import os; print(os.getcwd())'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
OSError: [Errno 2] No such file or directory
```

Reply:
yeah hadn't figured out exactly what was going on there yet...